src/rsz/src/RepairDesign.cc

```diff
@@ -1791,86 +1810,223 @@ void RepairDesign::makeRepeater(const char* reason,
   // between the driver and the loads changing the net as the repair works its
   // way from the loads to the driver.
 
-  Net *net = nullptr, *in_net;
+  Net* load_net = nullptr;
+  dbNet* load_db_net = nullptr;           // load net, flat
+  odb::dbModNet* load_mod_net = nullptr;  // load net, hierarchical
+
   bool preserve_outputs = false;
+  bool top_primary_input = false;
+  bool top_primary_output = false;
+
+  // Determine the type of the load
+  // primary input/ primary output/ dont touch
+
+  int number_of_valid_loads = 0;
+
   for (const Pin* pin : load_pins) {
+    load_db_net = db_network_->flatNet(pin);
+    load_mod_net = db_network_->hierNet(pin);
+
+    Instance* inst = network_->instance(pin);
+    if (resizer_->dontTouch(inst)) {
+      continue;
+    }
+    number_of_valid_loads++;
+
     if (network_->isTopLevelPort(pin)) {
-      net = network_->net(network_->term(pin));
+      load_net = network_->net(network_->term(pin));
+      db_network_->staToDb(load_net, load_db_net, load_mod_net);
+      load_db_net = db_network_->flatNet(pin);
       if (network_->direction(pin)->isAnyOutput()) {
         preserve_outputs = true;
+        top_primary_output = true;
+        break;
+      }
+      if (network_->direction(pin)->isInput()) {
```

Comment:
How can an input pin be a load pin?  The old code used hasInputPort which checks all of the net's connections.